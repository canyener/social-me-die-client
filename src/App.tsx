import React, { Component } from 'react'
import { Header, Icon, List } from 'semantic-ui-react'

import './App.css'
import axios from 'axios'

class App extends Component {
  state = {
    values: []
  }

  componentDidMount() {
    axios.get('http://localhost:5000/api/values').then(response => {
      this.setState(() => ({ values: response.data }))
    })
  }

  render() {
    return (
      <div>
        <Header as='h2'>
          <Icon name='users' />
          <Header.Content>Social Me Die</Header.Content>
        </Header>
        <List>
          {
            this.state.values.map((value: any) => 
              <List.Item key={value.id}>
                {value.name}
              </List.Item>)
          }
        </List>
      </div>
    )
  }
}

export default App
