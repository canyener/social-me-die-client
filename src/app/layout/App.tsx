import React, { useEffect, Fragment, useContext } from 'react'
import { Container } from 'semantic-ui-react'
import { observer } from 'mobx-react-lite'
import { Route, RouteComponentProps, withRouter } from 'react-router-dom'

import NavBar from '../../features/nav/NavBar'
import ActivityDashboard from '../../features/activities/dashboard/ActivityDashboard'
import LoadingComponent from './LoadingComponent'

import ActivityStore from '../stores/activityStore'
import HomePage from '../../features/home/HomePage'
import ActivityForm from '../../features/activities/form/ActivityForm'
import ActivityDetails from '../../features/activities/details/ActivityDetails'

const App: React.FC<RouteComponentProps> = ({ location }) => {

  const activityStore = useContext(ActivityStore)

  useEffect(() => {
    activityStore.loadActivities()
  }, [activityStore])

  if (activityStore.loadingInitial) return <LoadingComponent content="Loading Activities..." />

  return (
    <Fragment>
      <NavBar />
      <Container style={{ marginTop: '7em' }}>
        <Route exact path='/' component={HomePage} />
        <Route exact path='/activities' component={ActivityDashboard} />
        <Route path='/activities/:id' component={ActivityDetails} />
        <Route
          key={location.key}
          path={['/createActivity', '/manage/:id']}
          component={ActivityForm}
        />
      </Container>
    </Fragment>
  )
}

export default withRouter(observer(App))
