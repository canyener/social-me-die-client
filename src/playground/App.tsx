import React, { Component } from 'react'
import { Header, Icon, List } from 'semantic-ui-react'

import axios from 'axios'

import { IActivity } from '../app/models/activity'

interface IState {
  activities: IActivity[]
}

class App extends Component<{}, IState> {
  readonly state : IState = {
    activities: []
  }

  componentDidMount() {
    axios.get<IActivity[]>('http://localhost:5000/api/activities').then(response => {
      this.setState(() => ({ 
        activities: response.data 
      }))
    })
  }

  render() {
    return (
      <div>
        <Header as='h2'>
          <Icon name='users' />
          <Header.Content>Social Me Die</Header.Content>
        </Header>
        <List>
          {
            this.state.activities.map((activity) => 
              <List.Item key={activity.id}>
                {activity.title}
              </List.Item>)
          }
        </List>
      </div>
    )
  }
}

export default App
