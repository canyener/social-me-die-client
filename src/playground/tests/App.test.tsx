import React from 'react'
import { shallow } from 'enzyme'

import App from '../App'

const flushPromises = () => new Promise(resolve => setImmediate(resolve))

//const mockedAxios = axios as jest.Mocked<typeof axios>

test('Should render App correctly', () => {
  const wrapper = shallow(<App />)
  expect(wrapper).toMatchSnapshot()
})

test('Should render App correctly with data', async () => {
  const wrapper = shallow(<App />)
  await flushPromises()
  wrapper.update()
  expect(wrapper).toMatchSnapshot()
})
