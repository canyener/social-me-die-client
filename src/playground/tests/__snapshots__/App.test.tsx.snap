// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render App correctly 1`] = `
<div>
  <Header
    as="h2"
  >
    <Icon
      as="i"
      name="users"
    />
    <HeaderContent>
      Social Me Die
    </HeaderContent>
  </Header>
  <List />
</div>
`;

exports[`Should render App correctly with data 1`] = `
<div>
  <Header
    as="h2"
  >
    <Icon
      as="i"
      name="users"
    />
    <HeaderContent>
      Social Me Die
    </HeaderContent>
  </Header>
  <List>
    <ListItem
      key="1"
    >
      Fake Activity 1
    </ListItem>
    <ListItem
      key="2"
    >
      Fake Activity 2
    </ListItem>
  </List>
</div>
`;
